src/cfg/to_structured.rs

```diff
@@ -76,20 +84,30 @@ impl<'a> StructuredCfgBuilder<'a> {
         self.node_within(node, merge_nodes)
     }
 
-    fn node_within(&mut self, node: NodeIndex, merge_nodes: Vec<NodeIndex>) -> StructuredBlock {
+    fn node_within(
+        &mut self,
+        node: NodeIndex,
+        merge_nodes: Vec<NodeIndex>,
+    ) -> Option<StructuredBlock> {
         if node == self.cfg.exit {
-            return StructuredBlock::Basic(Box::new(self.cfg.graph[node].clone()));
+            return Some(StructuredBlock::Basic(Box::new(
+                self.cfg.graph[node].clone(),
+            )));
         }
 
         let edges = self
             .cfg
             .graph
             .edges_directed(node, petgraph::Direction::Outgoing)
             .collect::<Vec<_>>();
+        assert!(!edges.is_empty(), "only exit nodes should have empty edges");
+
         match merge_nodes.as_slice() {
             [] => {
-                StructuredBlock::Sequence(vec![
-                    StructuredBlock::Basic(Box::new(self.cfg.graph[node].clone())),
+                let v = vec![
+                    Some(StructuredBlock::Basic(Box::new(
```

Comment:
When do_branch returns None, on line 116 you can return this 
that way this function doesn't return an option